### PR TITLE
Allow expiration date to be today

### LINKED ISSUE #36569: default expire date config parameters can be set to 0 days
### Steps to reproduce
1. set `shareapi_expire_after_n_days_user_share` or `shareapi_expire_after_n_days_group_share` to 0
2. send a sharing API  request to create a share

### Expected behaviour
The parameter is not allowed to be zero.

### Actual behaviour
The share is created, and expires today

### Server configuration
Current core with feature branch `feature/user_group_share_expiration` PR #36531

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
The expiration date validation code had some issues when the server was not in UTC. 

The proposed expiration date was in a `DateTime` object that had the time-zone of the server - e.g. it represented `2020-01-27:00:00:00+01:00`. 

The date-time was being compared with "current day" - a new default `DateTime` object that had the time zeroed, but that object was in UTC time - e.g. `2020-01-27:00:00:00+00:00` - but in the local timezone that is actually `2020-01-27:01:00+01:00`

And so the proposed expiration date was 1 hour before the "current day" and the request would return "Expiration date is in the past".

- adjust the webUI JS date-picker so it allows "today"  to be selected
- add API acceptance tests that set share expiration to "today" for various combinations of default/enforced expiration

## Related Issue
- Fixes #36569 

## How Has This Been Tested?
CI an locally using the date-picker in webUI to choose "today"

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: core/js/sharedialoglinkexpirationview.js [hunk 1]
```diff
@@ -116,7 +116,7 @@
 			var minDate = new Date();
 			var maxDate = null;
 			// min date should always be the next day
-			minDate.setDate(minDate.getDate()+1);
+			minDate.setDate(minDate.getDate());
```
Comment (individual-it) [inline core/js/sharedialoglinkexpirationview.js]: what does this thing even do? get the date and set it to it self?

**Comment by phil-davis** [comment]
@micbar this is adjusting (fixing) the sharing API so it reliably accepts an expiration date of "today" for a user/group share. And adjusts the webUI date picker so that it is possible in the webUI to actually choose "today".

Please select an appropriate person to review the JS and PHP code changes.

@individual-it please review the acceptance test changes.

**Review by individual-it — DISMISSED** [review]

File: core/js/sharedialoglinkexpirationview.js (hunk 1 above)
Comment (phil-davis) [reply core/js/sharedialoglinkexpirationview.js]: Maybe it now does nothing. I claim zero JS knowledge! I just removed the `+1`
And tthe comment about is now rubbish - I better at least adjust/remove that!

Comment (phil-davis) [reply core/js/sharedialoglinkexpirationview.js]: removed this line and the comment above

**Comment by micbar** [comment]
@phil-davis How would this affect the clients, if they don't set a value because they know nothing about expiration date?

**Comment by phil-davis** [comment]
> @phil-davis How would this affect the clients, if they don't set a value because they know nothing about expiration date?

This PR only adjusts the existing code for user/group share expiry that is already in master for 10.4 There should be no effect on clients by any of this expiry stuff. If the default expiry is not set, then clients can optionally supply an expiry date (that would be for the clients to add in future), but when not mentioning a expiry date then no expiry date will be set. If a default expiry date is enabled (and also if enforced...) then the expiry date happens by default without the client needing to do anything.

**Review by individual-it — COMMENTED** [review]
tests look OK

**Review by micbar — APPROVED** [review]

**Pull request merged**